### PR TITLE
Refactor analysis vii

### PR DESCRIPTION
**Pull request by mschwoer**
Feature drop:
allow to save also statistical analysis, allow to download the resulting data.

Some frontend eye candy, and a refactoring of the frontend methods to be more DRY

### CODE REVIEW

File: alphastats/gui/pages/03_Preprocessing.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
             # TODO show more info about the preprocessing steps
             display_preprocessing_info(dataset.preprocessing_info)
 
-        if c12.button("Reset all Preprocessing steps", key="_reset_preprocessing"):
+        if c12.button("❌ Reset all Preprocessing steps", key="_reset_preprocessing"):
```
Comment (boopthesnoot) [inline alphastats/gui/pages/03_Preprocessing.py:44]: :100: :ok_hand:

File: alphastats/gui/pages/05_LLM.py [hunk 2]
```diff
@@ -216,7 +216,9 @@ def llm_chat(llm_integration: LLMIntegration, show_all: bool = False):
             for artifact in message["artifacts"]:
                 if isinstance(artifact, pd.DataFrame):
                     st.dataframe(artifact)
-                elif "plotly" in str(type(artifact)):
+                elif "plotly" in str(
+                    type(artifact)
+                ):  # TODO can there be non-plotly types here
```
Comment (boopthesnoot) [inline alphastats/gui/pages/05_LLM.py:221]: for now i don't think so

File: alphastats/gui/pages/05_LLM.py [hunk 3]
```diff
@@ -88,9 +88,9 @@ def llm_config():
     st.info("Create a Volcano plot first using the 'Analysis' page.")
     st.stop()
 
-volcano_plot, parameter_dict = st.session_state[StateKeys.LLM_INPUT]
+volcano_plot, plot_parameters = st.session_state[StateKeys.LLM_INPUT]
```
Comment (JuliaS92) [inline alphastats/gui/pages/05_LLM.py:91]: wouldn't this be analysis parameters, if we ever generalize to all analyses being interpretable by LLM?

File: alphastats/gui/utils/analysis.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 
 class PlottingOptions(metaclass=ConstantsClass):
-    """Keys for the plotting options."""
+    """Keys for the plotting options, the order determines order in UI."""
```
Comment (JuliaS92) [inline alphastats/gui/utils/analysis.py:14]: Put Volcano to the top.

File: alphastats/gui/utils/analysis_helper.py [hunk 5]
```diff
@@ -1,114 +1,194 @@
...
+    """Show a button to download a figure."""
 
     buffer = io.BytesIO()
 
     try:  # plotly
-        plot.write_image(file=buffer, format=format)
-    except AttributeError:
-        plot.savefig(buffer, format=format)
+        plot.write_image(file=buffer, format=file_format)
+    except AttributeError:  # TODO figure out what else "plot" can be
```
Comment (JuliaS92) [inline alphastats/gui/utils/analysis_helper.py:112]: seaborn apparently

File: alphastats/gui/utils/analysis_helper.py [hunk 6]
```diff
@@ -1,114 +1,194 @@
...
 
 
-def download_figure(method, plot, format):
-    """
-    download plotly figure
-    """
+def _show_buttons_download_figure(analysis_result: PlotlyObject, name: str) -> None:
+    """Show buttons to download figure as .pdf or .svg."""
+    _show_button_download_figure(analysis_result, name, "pdf")
+    _show_button_download_figure(analysis_result, name, "svg")
```
[the remark this answers is not in the document]
Comment (JuliaS92) [reply alphastats/gui/utils/analysis_helper.py:98]: rerendering with svg as renderer could be a method of PlotlyObject to invoke prior to saving as svg

**Review by JuliaS92 — APPROVED** [review]

File: alphastats/gui/pages/05_LLM.py (hunk 3 above)
Comment (mschwoer) [reply alphastats/gui/pages/05_LLM.py:91]: yes but for now it's tied to the volcano, that's we I chose to name it like so

**Pull request merged**